configParser.js

```diff
@@ -59,6 +59,14 @@ if (mongoOpts.ssl) {
   mongoOpts.sslCert = fs.readFileSync(mongoOpts.sslCert, 'utf8');
 }
 
+//warn user if their collection names are too large to help debug errors in configuration
+const COLL_NAME_MAX_LENGTH = 110;
+CONFIG.mongo.collections.forEach((collection) => {
+  if(collection.length > COLL_NAME_MAX_LENGTH){
+    console.log(`WARNING: The collection ${collection} is ${collection.length - COLL_NAME_MAX_LENGTH} characters too large.  Mongo does not allow names to be larger that 120 bytes.(https://docs.mongodb.com/manual/reference/limits/#Restriction-on-Collection-Names)`);
```

Comment:
The 110 character limit is intriguing. It appears the [actual limit is 120 bytes](https://docs.mongodb.com/manual/reference/limits/#Restriction-on-Collection-Names) for a collection namespace, database name and `.` delimiter inclusive.

Perhaps if we're going to go through the effort of preventing people from foot gunning, we should be precise:

```
const NAMESPACE_MAX_LENGTH = 120;

CONFIG.mongo.collections.forEach((collection) => {
  let ns_length = db.length + '.'.length + collection.length;
  if(ns_length > NAMESPACE_MAX_LENGTH){
    console.log(`WARNING: The collection ${collection}'s namespace is ${ns_length - NAMESPACE_MAX_LENGTH} characters too large.  Mongo does not allow names to be larger that 120 bytes.(https://docs.mongodb.com/manual/reference/limits/#Restriction-on-Collection-Names)`);
  }
});

```

Reply:
Oh, you already had the 120 byte limit in there in the error.

Reply:
when we tested creating a collection with a long name it maxed at 116 characters.  When we tried 117 it gave an error that it was too long.  I don't understand how that correlates to 120 bytes but we decided that just checking at 110 was reasonable.  It is only a warning to help debugging.  We will accept any length.

Reply:
Sounds good to me, my suggestion was mostly in jest. Primarily an outlet for internal musings regarding the seemingly arbitrary MongoDB namespace max length. Maybe they are using UTF-16, maybe it's blood magic, maybe it sounded good on paper.